generators/client/templates/angular/src/test/javascript/spec/app/shared/user/_user.service.spec.ts

```diff
@@ -16,97 +16,78 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 
 describe('Service Tests', () => {
 
     describe('User Service', () => {
         let service: UserService;
+        let httpMock;
 
-        beforeEach(async(() => {
+        beforeEach(() => {
             TestBed.configureTestingModule({
+                imports: [
+                    HttpClientTestingModule
+                ],
                 providers: [
-                    {
-                        provide: ConnectionBackend,
-                        useClass: MockBackend
-                    },
-                    {
-                        provide: RequestOptions,
-                        useClass: BaseRequestOptions
-                    },
-                    Http,
                     JhiDateUtils,
                     UserService
                 ]
             });
 
             service = TestBed.get(UserService);
+            httpMock = TestBed.get(HttpTestingController);
+        });
 
-            this.backend = TestBed.get(ConnectionBackend) as MockBackend;
-            this.backend.connections.subscribe((connection: any) => {
-                this.lastConnection = connection;
-            });
-        }));
+        afterEach(() => {
+            httpMock.verify();
+        })
 
         describe('Service methods', () => {
             it('should call correct URL', () => {
-                service.find('user').subscribe(() => {});
-                const resourceUrl = SERVER_API_URL + '<%- apiUaaPath %>api/users';
+                service.find('user');
 
-                expect(this.lastConnection).toBeDefined();
-                expect(this.lastConnection.request.url).toEqual(`${resourceUrl}/user`);
+                const req  = httpMock.expectOne({ method: 'GET' });
+                const resourceUrl = SERVER_API_URL + '<%- apiUaaPath %>api/users';
+                expect(req.request.url).toEqual(`${resourceUrl}/user`);
             });
             it('should return User', () => {
 
-                let entity: User;
-                service.find('user').subscribe((_entity: User) => {
-                    entity = _entity;
+                service.find('user').subscribe((received) => {
+                    expect(received.body.login).toEqual('user');
                 });
 
-                this.lastConnection.mockRespond(new Response(new ResponseOptions({
-                    body: JSON.stringify(new User(1, 'user')),
-                })));
-
-                expect(entity).toBeDefined();
-                expect(entity.login).toEqual('user');
+                const req = httpMock.expectOne({ method: 'GET' });
+                req.flush(new User(1, 'user'));
             });
 
             it('should return Authorities', () => {
 
-                let authorities;
                 service.authorities().subscribe((_authorities) => {
-                    authorities = _authorities;
+                    expect(_authorities).toEqual(['ROLE_USER', 'ROLE_ADMIN']);
```

Comment:
this will fail in cassandra hence this test should be skipped when DB is cassandra

Reply:
excluded the test if cassandra is used